packages/components/src/components/form/Switch/Switch.tsx

```diff
@@ -18,46 +24,39 @@ const Container = styled.div<Pick<SwitchProps, 'isChecked' | 'isDisabled' | 'isS
     transition: background 0.25s ease 0s;
     cursor: ${({ isDisabled }) => !isDisabled && 'pointer'};
     box-sizing: border-box;
+    border: 1px solid ${({ theme, isAlert }) => `${isAlert ? theme.STROKE_RED : 'transparent'}`};
 
-    ${({ isDisabled }) =>
+    button {
+        box-shadow: 0px 2px 4px 0px ${({ theme }) => theme.BOX_SHADOW_BLACK_5};
+        opacity: ${({ isDisabled }) => isDisabled && 0.66};
+    }
+
+    ${({ isDisabled, theme, isChecked }) =>
         !isDisabled &&
         css`
-            :hover,
-            :focus-within {
-                button {
-                    box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 1);
-                }
+            :hover {
+                background: ${isChecked ? theme.BG_GREEN_HOVER : theme.STROKE_GREY_HOVER};
             }
-
-            :active {
-                button {
-                    box-shadow: none;
-                }
+            :focus-within {
```

Comment:
`:focus-within:has(:focus-visible)` so that it doesn't gain the outline when the Handle is clicked